ledger/src/sigverify_shreds.rs

```diff
@@ -588,41 +590,46 @@ mod tests {
         solana_logger::setup();
         let mut batch = PinnedPacketBatch::default();
         let cache = RwLock::new(LruCache::new(/*capacity:*/ 128));
-        let mut shred = Shred::new_from_data(
-            slot,
-            0xc0de,
-            0xdead,
-            &[1, 2, 3, 4],
-            ShredFlags::LAST_SHRED_IN_SLOT,
+        let shredder = Shredder::new(slot, slot.saturating_sub(1), 0, 0).unwrap();
+        let keypair = Keypair::new();
+        let reed_solomon_cache = ReedSolomonCache::default();
+        let (shreds, _) = shredder.entries_to_merkle_shreds_for_tests(
+            &keypair,
+            &[],
+            true,
+            Some(Hash::default()),
             0,
             0,
-            0xc0de,
+            &reed_solomon_cache,
+            &mut ProcessShredsStats::default(),
         );
-        let keypair = Keypair::new();
-        shred.sign(&keypair);
-        batch.resize(1, Packet::default());
-        batch[0].buffer_mut()[..shred.payload().len()].copy_from_slice(shred.payload());
-        batch[0].meta_mut().size = shred.payload().len();
+        batch.resize(shreds.len(), Packet::default());
```

Comment:
I haven't looked at `PacketBatch` / `Packet` in a while, but we were trying to get rid of `Packet::default()` as we were doing some stuff in there that is technically UB. I'm not sure where we ended up there, @vadorovsky might know off hand or if not, I can go dig.

I'd say no action on this item required yet

Reply:
Yeah I'm aware of what he is working on, whatever UB might creep up there will not be a problem for this test=)

Reply:
> whatever UB might creep up there will not be a problem for this test=)

I mean yes, that UB luckily hasn't bit us yet. My comment is more suggesting that we should avoid adding new instances of the function with UB.

That being said, I do now see that `batch.resize(x, Packet::default())` was previously in this test. So in that sense, we are not increasing the usage count and I'm ok with that. Going to mark this convo resolved